src/dc_chat.c

```diff
@@ -1984,9 +1984,65 @@ static int last_msg_in_chat_encrypted(dc_sqlite3_t* sql, uint32_t chat_id)
 }
 
 
+static int get_parent_mime_headers(const dc_chat_t* chat,
+                                   char**           parent_rfc724_mid,
+                                   char**           parent_in_reply_to,
+                                   char**           parent_references)
+{
+	int           success = 0;
+	sqlite3_stmt* stmt = NULL;
+
+	if (chat==NULL
+	 || parent_rfc724_mid==NULL || parent_in_reply_to==NULL || parent_references==NULL) {
+		goto cleanup;
+	}
+
+	// use the last messsage of another user in the group as the parent
+	stmt = dc_sqlite3_prepare(chat->context->sql,
+		"SELECT rfc724_mid, mime_in_reply_to, mime_references"
+		" FROM msgs"
+		" WHERE timestamp=(SELECT max(timestamp) FROM msgs WHERE chat_id=? AND from_id!=?);");
+	sqlite3_bind_int  (stmt, 1, chat->id);
+	sqlite3_bind_int  (stmt, 2, DC_CONTACT_ID_SELF);
+	if (sqlite3_step(stmt)==SQLITE_ROW) {
+		*parent_rfc724_mid  = dc_strdup((const char*)sqlite3_column_text(stmt, 0));
+		*parent_in_reply_to = dc_strdup((const char*)sqlite3_column_text(stmt, 1));
+		*parent_references  = dc_strdup((const char*)sqlite3_column_text(stmt, 2));
+		success = 1;
+	}
+	sqlite3_finalize(stmt);
+	stmt = NULL;
+
+	if (!success) {
+		// there are not messages of other users - use the first message if SELF as parent
+		stmt = dc_sqlite3_prepare(chat->context->sql,
+			"SELECT rfc724_mid, mime_in_reply_to, mime_references"
+			" FROM msgs"
+			" WHERE timestamp=(SELECT min(timestamp) FROM msgs WHERE chat_id=? AND from_id==?);");
```

Comment:
again i don't understand why this is constrained by from_id. probably missing something?

Reply:
see above.